compiler/rustc_infer/src/infer/mod.rs

```diff
@@ -371,6 +371,7 @@ pub enum ValuePairs<'tcx> {
     Types(ExpectedFound<Ty<'tcx>>),
     Regions(ExpectedFound<ty::Region<'tcx>>),
     Consts(ExpectedFound<&'tcx ty::Const<'tcx>>),
```

Comment:
I was hoping that as a next step we could remove type and const entirely from this enum.Do you plan on tackling that? If not, can you please open an issue

Reply:
yea, I was just worried about diverging logic, and as the diagnostic regression shows, people are matching on the specific variant for diagnostics.

So, if you feel like messing with all the use sites of `::Ty`, nix `ValuePair::Ty`, otherwise, just fix the diagnostics regression and open a cleanup issue

Reply:
I went thru, wasn't that many different sites so I just replaced it